cli/azd/cmd/up.go

```diff
@@ -52,68 +52,46 @@ func newUpCmd() *cobra.Command {
 }
 
 type upAction struct {
-	flags                      *upFlags
-	env                        *environment.Environment
-	projectConfig              *project.ProjectConfig
-	packageActionInitializer   actions.ActionInitializer[*packageAction]
-	provisionActionInitializer actions.ActionInitializer[*provisionAction]
-	deployActionInitializer    actions.ActionInitializer[*deployAction]
-	console                    input.Console
-	runner                     middleware.MiddlewareContext
-	prompters                  prompt.Prompter
-	provisioningManager        *provisioning.Manager
-	importManager              *project.ImportManager
+	flags               *upFlags
+	console             input.Console
+	env                 *environment.Environment
+	projectConfig       *project.ProjectConfig
+	provisioningManager *provisioning.Manager
+	importManager       *project.ImportManager
+	workflowRunner      *WorkflowRunner
+}
+
+var defaultUpWorkflow = &workflow.Workflow{
+	Name: "up",
+	Steps: []*workflow.Step{
+		{AzdCommand: workflow.Command{Args: []string{"package", "--all"}}},
+		{AzdCommand: workflow.Command{Args: []string{"provision"}}},
+		{AzdCommand: workflow.Command{Args: []string{"deploy", "--all"}}},
+	},
 }
 
 func newUpAction(
 	flags *upFlags,
+	console input.Console,
 	env *environment.Environment,
 	_ auth.LoggedInGuard,
 	projectConfig *project.ProjectConfig,
-	packageActionInitializer actions.ActionInitializer[*packageAction],
```

Comment:
I do wonder if long term we should move away from the ActionInitializer stuff we have here in favor of just calling via the Cobra layer for the few places we still use the actions.ActionInitializer concept.  Something for us to look at over time, perhaps.

Reply:
yes, we could just create & execute a workflow object since internally this will bootstrap from the cobra commands.